### PR TITLE
LPS-149497 Add externalReferenceCode to Comments and include endpoints by externalReferenceCode

### PR DESCRIPTION
**Pull request by ccorreagg**
This PR contains the code to include in the Headless Delivery API the following functionality:

- Expose **externalReferenceCode** field in the already existing **Comment** element
- Expose new endpoints fo manage Comments by all of its possible parents: BlogPostings, another Comments, Documents and StructuredContents.

As PM has decided, as every Comment is associated to a one of the already referenced parents, the endpoints are exposed based on the **siteId** and the **externalReferenceCode** of the parent for all the cases. The methods involved are **PUT** (upsert), **GET** and **DELETE** as in the rest of the Headless Delivery API for ERC functionality.

- Another comment:

```
"/sites/{siteId}/comments/by-external-reference-code/{parentCommentExternalReferenceCode}/comments/by-external-reference-code/{externalReferenceCode}":
    delete:
        [...]
    get:
        [...]
    put:
        [...]
```

- BlogPosting:

```
"/sites/{siteId}/blog-postings/by-external-reference-code/{blogPostingExternalReferenceCode}/comments/by-external-reference-code/{externalReferenceCode}":
    delete:
        [...]
    get:
        [...]
    put:
        [...]
```

- Document:

```
"/sites/{siteId}/documents/by-external-reference-code/{documentExternalReferenceCode}/comments/by-external-reference-code/{externalReferenceCode}":
    delete:
        [...]
    get:
        [...]
    put:
        [...]
```

- StructuredContent:

```
"/sites/{siteId}/structured-contents/by-external-reference-code/{structuredContentExternalReferenceCode}/comments/by-external-reference-code/{externalReferenceCode}":
    delete:
        [...]
    get:
        [...]
    put:
        [...]
```
---
### How to test

**Exposure of **externalReferenceCode** field:**

First of all, it is necessary to create a parent for the Comment, for example, a BlogPosting:

```
curl -X POST 'http://localhost:8080/o/headless-delivery/v1.0/sites/20123/blog-postings' \
     -H 'accept: application/json' \
     -H 'Content-Type: application/json' \
     -u 'test@liferay.com:test' \
     -d '{
             "headline": "Blog posting headline",
             "articleBody": "Blog posting body"
         }'
```

Once created, with the id returned, it is possible to create a Comment including the externalReferenceCode:

```
curl -X POST 'http://localhost:8080/o/headless-delivery/v1.0/blog-postings/48059/comments' \
     -H 'accept: application/json' \
     -H 'Content-Type: application/json' \
     -u 'test@liferay.com:test' \
     -d '{
             "externalReferenceCode": "ERC001",
             "text": "This is my comment"
         }'
```

The response already includes the set externalReferenceCode. It is also possible to get the comment and its externalReferenceCode by calling the GET method with the comment identifier:

```
curl -X GET 'http://localhost:8080/o/headless-delivery/v1.0/comments/48069' \
     -u 'test@liferay.com:test'
```

**New endpoints by externalReferenceCode:**

First of all, it is necessary to create a parent for the Comment, for example, a BlogPosting with the externalReferenceCode field:

```
curl -X POST 'http://localhost:8080/o/headless-delivery/v1.0/sites/20123/blog-postings' \
     -H 'accept: application/json' \
     -H 'Content-Type: application/json' \
     -u 'test@liferay.com:test' \
     -d '{
             "headline": "Blog posting headline 2",
             "articleBody": "Blog posting body 2",
             "externalReferenceCode": "MYBLOG01"
         }'
```
Once created, it is possible to create a Comment associated to the BlogPosting by its externalReferenceCode:

```
curl -X PUT 'http://localhost:8080/o/headless-delivery/v1.0/sites/20123/blog-postings/by-external-reference-code/MYBLOG01/comments/by-external-reference-code/MYCOMMENT01' \
     -H 'accept: application/json' \
     -H 'Content-Type: application/json' \
     -u 'test@liferay.com:test' \
     -d '{
             "externalReferenceCode": "MYCOMMENT01",
             "text": "This is my comment"
         }'
```
The response already includes the set externalReferenceCode. It is also possible to get the comment by calling the GET method with the ERC of the BlogPosting and the Comment:

```
curl -X GET 'http://localhost:8080/o/headless-delivery/v1.0/sites/20123/blog-postings/by-external-reference-code/MYBLOG01/comments/by-external-reference-code/MYCOMMENT01' \
     -u 'test@liferay.com:test'
```

### CODE REVIEW

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:test:relevant

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/CommentResourceImpl.java [hunk 1]
```diff
@@ -191,6 +252,94 @@ public EntityModel getEntityModel(MultivaluedMap multivaluedMap) {
...
 	}
 
+	@Override
+	public Comment
+			getSiteBlogPostingByExternalReferenceCodeBlogPostingExternalReferenceCodeCommentByExternalReferenceCode(
+				Long siteId, String blogPostingExternalReferenceCode,
+				String externalReferenceCode)
+		throws Exception {
+
+		BlogsEntry blogsEntry = _getBlogsEntry(
```
Comment (javierdearcos) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/CommentResourceImpl.java:262]: Very nice work making that each use case follows the same structure in the code: _getEntity, _getComment, checkpermission, return comment. Same with the other use cases 👏

**Review by javierdearcos — APPROVED** [review]
Great job!

A little bit more and we would have broken the SourceFormatter with the long method names 😂

**Comment by javierdearcos** [comment]
ci:forward

**Comment by ccorreagg** [comment]
> Great job!
> 
> A little bit more and we would have broken the SourceFormatter with the long method names joy

:joy: Those names are not beautiful at all :no_good:

**Pull request closed without merging**